### PR TITLE
[WIP] Bigbang

### PR DESCRIPTION
**Pull request by Isammoc**
Note for unwanted (for now) reviewers:
 * I set the version to 3.0.0 (major upgrade) because I foresee that I will break a lot of things (intended)
 * This PR is currently very small, only to ensure that the tests will play well

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache SBT
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.ivy2/cache
+            ~/.sbt
+          key: ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('**/build.properties') }}
+      - uses: olafurpg/setup-scala@v10
```
Comment (Isammoc) [inline .github/workflows/test.yml:31]: I'd prefer to use java and have an sbt wrapper

File: src/sbt-test/scalafix/none/build.sbt [hunk 2]
```diff
@@ -0,0 +1,24 @@
+scalaVersion := "2.12.3"
+
+def assertContentsEqual(file: File, expected: String): Unit = {
```
Comment (Isammoc) [inline src/sbt-test/scalafix/none/build.sbt:3]: Not used?

File: src/sbt-test/scalafmt/automated/build.sbt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+scalaVersion := "2.12.3"
+scalafixDependencies in ThisBuild += "org.scalameta" % "sbt-scalafmt" % "2.3.4"
+
+lazy val root = (project in file("."))
+  .enablePlugins(GatlingAutomatedScalafmtPlugin)
+  .settings(
+    version := "0.1",
+    name := "scalafmt/automated"
+  )
```
Comment (Isammoc) [inline src/sbt-test/scalafmt/automated/build.sbt:9]: EOL

File: src/main/scala/io/gatling/build/Milestone.scala [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  lazy val milestoneQualifierPrefix = "-M"
+  lazy val milestoneFormatterPattern = "yyyyMMddhhmmss"
+  lazy val milestoneFormatter = new SimpleDateFormat(milestoneFormatterPattern)
+
+  class MilestoneVersion(version: Version) {
+    def isMilestone: Boolean =
+      version.qualifier.exists { qualifier =>
+        qualifier.length == milestoneFormatterPattern.length + milestoneQualifierPrefix.length &&
+          qualifier.startsWith(milestoneQualifierPrefix) &&
+          Try(milestoneFormatter.parse(qualifier.substring(2))).isSuccess
```
Comment (Isammoc) [inline src/main/scala/io/gatling/build/Milestone.scala:22]:
```suggestion
          Try(milestoneFormatter.parse(qualifier.substring(milestoneQualifierPrefix.length))).isSuccess
```

File: src/main/scala/io/gatling/build/Milestone.scala [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+import sbtrelease.Version
+
+import scala.language.implicitConversions
+import scala.util.Try
+
+object Milestone {
+
+  lazy val milestoneQualifierPrefix = "-M"
+  lazy val milestoneFormatterPattern = "yyyyMMddhhmmss"
```
Comment (Isammoc) [inline src/main/scala/io/gatling/build/Milestone.scala:14]: Constants should start with Uppercase

File: src/main/scala/io/gatling/build/Milestone.scala [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      }
+
+    def asMilestone: Version = {
+      val qualifier = milestoneQualifierPrefix + milestoneFormatter.format(new Date())
+      version.copy(qualifier = Some(qualifier))
+    }
+  }
+
+  implicit def milestoneVersion(version: Version): MilestoneVersion =
+    new MilestoneVersion(version)
```
Comment (Isammoc) [inline src/main/scala/io/gatling/build/Milestone.scala:32]: `implicit class` ?

File: src/main/scala/io/gatling/build/ReleaseProcessKeys.scala [hunk 7]
```diff
@@ -1,49 +1,159 @@
 package io.gatling.build
 
-import scala.util.Properties
-
+import _root_.io.gatling.build.GatlingReleasePlugin.autoimport._
 import io.gatling.build.MavenPublishKeys.pushToPrivateNexus
-
 import sbt.Keys._
 import sbt._
+import _root_.io.gatling.build.Milestone._
```
Comment (Isammoc) [inline src/main/scala/io/gatling/build/ReleaseProcessKeys.scala:7]: organize imports?

File: src/main/scala/io/gatling/build/ReleaseProcessKeys.scala [hunk 8]
```diff
@@ -1,49 +1,159 @@
...
+          version => version.subversions.isDefinedAt(1) && version.subversions(1) == 0,
+          version => s"Cannot release a patch version when current version is minor (${version.string})"
+        ) +: commonReleaseSteps(commitVersion = true)) ++ Seq(
+          setNextVersion,
+          commitNextVersion,
+          pushChanges
+        )
+      case GatlingReleaseProcess.Milestone =>
+        setMilestoneReleaseVersion +: commonReleaseSteps(commitVersion = false)
+    }
```
Comment (Isammoc) [inline src/main/scala/io/gatling/build/ReleaseProcessKeys.scala:155]: Hard to follow what are the steps for the current release...

File: .github/workflows/test.yml (hunk 1 above)
Comment (tpetillot) [reply .github/workflows/test.yml:31]: TBH, I have no idea about those github workflows, could you explain ?

**Comment by tpetillot** [comment]
Cannot ask for @Isammoc review as he own the PR
